src/object_api.c

```diff
@@ -42,6 +42,10 @@ git_repository *git_commit_owner(const git_commit *obj)
 	return git_object_owner((const git_object *)obj);
 }
 
+int git_commit_dup(git_commit **out, git_commit *obj)
+{
+	return git_object_dup((git_object **)out, (git_object *)obj);
```

Comment:
This is really just :lipstick: so I'm not sure that's necessary.  For `obj` checking to ever find a false positive, then a caller would have done something like:

    git_tree *tree;
    git_commit *dup;
    git_commit_dup(&dup, (git_commit *)tree);

Which probably makes no sense.  Recall that `git_object_dup` and friends are internal-only, so this isn't anything that would help our callers.  But maybe I am missing the use case?